src/freenet/node/PeerLocation.java

```diff
@@ -1,14 +1,16 @@
 package freenet.node;
 
 import java.util.Arrays;
+import java.util.Set;
 
 import freenet.support.Logger;
 
 public class PeerLocation {
 	
 	/** Current location in the keyspace, or -1 if it is unknown */
 	private double currentLocation;
-	/** Current locations of our peer's peers */
+	/** Current sorted array of locations of our peer's peers. Must not be modified,
+	 may only be replaced entirely. */
 	private double[] currentPeersLocation;
```

Comment:
It would be an optimization indeed, moving another single check into the logarithmic complexity.

There is code (PeerManager.java#L1057) keeping track of whether our peer's location or a FOAF location was chosen. Including `currentLocation` here would break that.